### DISCUSSION TITLE
Questions about storing ES in Skywalking OAP

### DISCUSSION THREAD

**Discussion by liucx1201** [apache/skywalking#11802] [Q&A]
Skywalking is stored in the Elasticsearch cluster

ES has done hot(SSD) and cold  separation, and there is a label on the ES node: hot

May I ask how to add tags to Skywalking storage in ES for easy storage on hot nodes

**Comment by wu-sheng** [comment]
We don't have that concept from today implementation.
How that works? Do you need some kinds of parameters in the write bulk?

**Reply by liucx1201** [reply]
![es](https://github.com/apache/skywalking/assets/40434832/7d5028dd-dc88-415d-8989-dfe1e98da26d)
The index of es is: {"index.routing.allocation.require.temperature": "hot"}
Hope sw_segment* These indexes also add this label to segment, which stores the hot nodes of ES

**Reply by wu-sheng** [reply]
You could use the index setting to override the default configuration, such as `specificIndexSettings`
https://skywalking.apache.org/docs/main/next/en/setup/backend/storages/elasticsearch/#index-settings

**Reply by liucx1201** [reply]
SW_STORAGE_ES_SPECIFIC_INDEX_SETTINGS: "{\"index.routing.allocation.require.temperature\": \"hot\"}"
SW_STORAGE_ES_ADVANCED: "{\"index.routing.allocation.require.temperature\": \"hot\"}"

SW_STORAGE_ES_ADVANCED: "{\"number_of_shards\":\"3\"}"

SW_STORAGE_ES_SPECIFIC_INDEX_SETTINGS: "{\"metrics-all\":{\"routing.allocation.require.temperature\":\"hot\"},\"segment\":{\"routing.allocation.require.temperature\":\"hot\"}}"

These configurations have been tested, but none of them are effective

**Reply by wu-sheng** [reply]
The source codes are there, you could easily debug and test.

**Reply by wu-sheng** [reply]
Note, the index settings only take effect when the index is created the first time, it will not update the settings. The indices are expected to be in the product env, we would not take the risk to change product env settings.
